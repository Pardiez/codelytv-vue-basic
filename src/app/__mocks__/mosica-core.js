import { fakeGigsByDay } from './gigs-sample'
const mosicaCore = jest.genMockFromModule('mosica-core')

class GigService {
  retrieveNextGigs() {
    return Promise.resolve(fakeGigsByDay)
  }
}

mosicaCore.GigService = GigService
module.exports = mosicaCore
