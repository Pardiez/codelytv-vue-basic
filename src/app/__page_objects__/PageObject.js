export default class PageObject {
  constructor(wrapper) {
    this.wrapper = wrapper
  }

  async updateAsync() {
    await flushPromises()
    this.wrapper.update()
  }

  flushPromises() {
    return new Promise(resolve => setImmediate(resolve))
  }

  text() {
    return this.wrapper.text()
  }

  html() {
    return this.wrapper.html()
  }

  contains(text) {
    expect(this.text()).toContain(text)
  }

  matchSnapshot() {
    expect(this.wrapper.html()).toMatchSnapshot()
  }
}
