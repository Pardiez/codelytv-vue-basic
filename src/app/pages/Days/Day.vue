<template>
  <q-card v-if="!isLoading">
    <div class='day'>
      <q-card-title>{{dateLabel}}</q-card-title>

      <div v-for="gig in day.gigs">
        <Gig :gig ="gig" :onClick="goTo"/>
      </div>
    </div>
  </q-card>
</template>

<script>
  import { MosicaRouter } from '../../services/MosicaRouter'
  import { spanishFromIso } from '../../services/date-utils'

  export default {
    props: {day: Object, isLoading: Boolean},
    created() {
      this.mosicaRouter = new MosicaRouter(this.$router)
    },
    methods: {
      goTo(gig) {
        this.mosicaRouter.navigateToGig(gig.id)
      }
    },
    computed: {
      dateLabel() {
        return spanishFromIso(this.day.date)
      }
    }
  }
</script>

<style scoped>
  .q-card-title {
    color: black;
    font-weight: bold;
    font-size: 150%;
  }
</style>
