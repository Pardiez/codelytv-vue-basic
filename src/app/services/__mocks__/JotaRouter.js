const getParam = () => 'any id'
module.exports = {
  JotaRouter: () => {
    return { getParam }
  }
}
