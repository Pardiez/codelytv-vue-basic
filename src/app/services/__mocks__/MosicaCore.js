import { fakeGigsByDay } from './gigs-sample'

class GigService {
  retrieveNextGigs() {
    return Promise.resolve(fakeGigsByDay)
  }
}

class Matcher {}

module.exports = { GigService, Matcher }
