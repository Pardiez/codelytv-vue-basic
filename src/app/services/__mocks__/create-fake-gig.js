import faker from 'faker'
import { isoFuture, spanishFromIso } from '../date-utils'

export function createFakeGig(date) {
  return {
    id: faker.random.number().toString(),
    title: randomTitle(),
    description: faker.lorem.paragraph(),
    image_url: randomImageUrl(),
    lat_lng: `${faker.address.latitude()},${faker.address.longitude()}`,
    address: faker.address.streetAddress(),
    price: randomNumber(5, 100) + '€',
    place: randomPlace(),
    hour: randomNumber(4, 11) + 'PM',
    day: date
  }
}

function randomNumber(min, max) {
  return faker.random.number({min, max})
}

export function createFakeDay(date) {
  return {
    day: spanishFromIso(date),
    gigs: Array(randomNumber(1, 10)).fill().map(() => createFakeGig(date))
  }
}

export function createFakeDays() {
  return Array(randomNumber(3, 10)).fill().map((number, index) => {
    return createFakeDay(isoFuture(index))
  })
}

function randomImageUrl() {
  return imageUrls[randomNumber(0, imageUrls.length - 1)]
}

const imageUrls = [
  'https://i.pinimg.com/736x/0e/de/f8/0edef8dd970d352a95a8caeaa55aa81c--marina-dancing.jpg',
  'https://i.pinimg.com/564x/95/83/8b/95838bc6e43c5229b7a677a78c70e09a.jpg',
  'https://i.pinimg.com/564x/08/d5/dd/08d5ddd4bb0c8a59b7a4021ba87a38e8.jpg',
  'https://i.pinimg.com/564x/31/c7/4e/31c74e89d905b300e3832b439ce7ffc0.jpg',
  'https://i.pinimg.com/564x/97/0b/73/970b736b7d75062ab9a525ad5f3c3497.jpg',
  'https://i.pinimg.com/564x/cf/b7/b6/cfb7b6fc2000dfdbb9f4aeb8800f7e87.jpg',
  'https://i.pinimg.com/564x/59/bb/59/59bb5945d68e8d913b2130f3628949b9.jpg'
]

function randomPlace() {
  return places[randomNumber(0, places.length - 1)]
}

const places = [
  'Sala Razzmatazz',
  'Codely Headquarters',
  'Pabellón Principe Felipe',
  'Estadio Vicente Calderón',
  'Palacio Municipal de los Deportes Huesca',
  'Joy Slava',
  'Sala Z',
  'Estadio de Wembley',
  'CBGB New York',
  'Teatro Principal de Zaragoza'
]

function randomTitle() {
  return firstTerm[randomNumber(0, firstTerm.length - 1)] + ' ' +
  secondTerm[randomNumber(0, secondTerm.length - 1)] + ' ' +
  thirdTerm[randomNumber(0, thirdTerm.length - 1)]
}

const firstTerm = [
  'Grupo', 'Agrupación', 'Amigos', 'Folklore', 'Colectivo', 'Banda', 'Ronda', 'Rondalla', 'Escuela'
]

const secondTerm = [
  'Folclore', 'Baluarte', 'Danzantes', 'Joteros', 'Jota', 'Baile'
]

const thirdTerm = [
  'De Zaragoza', 'Aragonés', 'Turolense', 'Oscense'
]

// {
//   'schedule_id': 5280,
//   'schedule': [
//   {
//     'id': 5280,
//     'day': '2017-09-18',
//     'hour': '19:30',
//     'gig_id': 5927,
//     'created_at': '2017-09-07T15:57:35.419Z',
//     'updated_at': '2017-09-07T15:57:35.419Z'
//   }
// ]
// },
