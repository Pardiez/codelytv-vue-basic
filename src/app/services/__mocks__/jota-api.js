// function createGig(payload) {
//   return Promise.resolve(payload)
// }

import { fakeGigsByDay } from './gigs-sample'

const createGigStub = jest.fn((payload) => {
  return Promise.resolve(payload)
})

const retrieveDaysStub = jest.fn(() => {
  return Promise.resolve(fakeGigsByDay)
})

module.exports = { createGig: createGigStub, retrieveDays: retrieveDaysStub }
