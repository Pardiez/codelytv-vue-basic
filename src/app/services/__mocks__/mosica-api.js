// function createGig(payload) {
//   return Promise.resolve(payload)
// }

const createGigStub = jest.fn((payload) => {
  Promise.resolve(payload)
})

module.exports = { createGig: createGigStub }
