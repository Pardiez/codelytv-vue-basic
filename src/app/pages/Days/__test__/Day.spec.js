import Day from '@/app/pages/Days/Day.vue'
import PageObject from '../../../__page_objects__/PageObject'
import { fakeGigsByDay } from '../../../__mocks__/gigs-sample'

const FIRST_DAY = fakeGigsByDay[0]

describe('Day', () => {

  let page, wrapper
  beforeEach(async () => {
    wrapper = mountWithProps(Day, {day: FIRST_DAY, isLoading: false, onClick: jest.fn })
    page = new PageObject(wrapper)
    await page.updateAsync()
  })

  // Do we need tests to check onClick works as it is already tested in integration test (Days.spec)?
  // The same applies to isLoading

  it('matches full snapshot', async() => {
    page.matchSnapshot()
  })

  it('renders gig', async() => {
    page.contains(FIRST_DAY.day)
  })
})
