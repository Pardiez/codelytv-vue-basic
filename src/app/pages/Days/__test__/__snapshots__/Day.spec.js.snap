// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day matches full snapshot 1`] = `
<div class="q-list">
    <div class="day">
        <div class="q-list-header ">Lunes, 18 de Septiembre</div>
        <div>
            <div class="q-item q-item-division relative-position q-item-highlight">
                Anarchy in the JS - SCBCN
            </div>
        </div>
        <div>
            <div class="q-item q-item-division relative-position q-item-highlight">
                Vegan Microservices - Around the world
            </div>
        </div>
    </div>
</div>
`;
